### PR TITLE
Minor refactoring

### PR DESCRIPTION
**Pull request by graetzer**
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/2169/

### CODE REVIEW

File: arangod/RocksDBEngine/RocksDBVPackIndex.cpp [hunk 1]
```diff
@@ -640,88 +639,76 @@ Result RocksDBVPackIndex::insertInternal(transaction::Methods* trx,
...
       // The estimator is only useful if we are in a non-unique indexes
       TRI_ASSERT(!_unique);
       state->trackIndexInsert(_collection.id(), id(), it);
     }
-  }
-
-  if (res == TRI_ERROR_ARANGO_UNIQUE_CONSTRAINT_VIOLATED) {
-    LocalDocumentId documentId = RocksDBValue::documentId(existing);
+  } else if (res.is(TRI_ERROR_ARANGO_UNIQUE_CONSTRAINT_VIOLATED)) {
+    // find conflicing document
```
Comment (jsteemann) [inline arangod/RocksDBEngine/RocksDBVPackIndex.cpp]:
```suggestion
    // find conflicting document
```

File: arangod/RocksDBEngine/RocksDBVPackIndex.cpp [hunk 2]
```diff
@@ -640,88 +639,76 @@ Result RocksDBVPackIndex::insertInternal(transaction::Methods* trx,
...
+    [&](LocalDocumentId const&, VPackSlice doc) {
+      existingKey = transaction::helpers::extractKeyFromDocument(doc).copyString();
     });
     TRI_ASSERT(success);
 
     if (mode == OperationMode::internal) {
-      return IndexResult(res, std::move(existingKey));
+      res.resetErrorMessage(std::move(existingKey));
+    } else {
+      addErrorMsg(res);
```
Comment (jsteemann) [inline arangod/RocksDBEngine/RocksDBVPackIndex.cpp]: can we add the existing key to the error message here? I think it will be missing now.

**Comment by jsteemann** [comment]
Additional question: do we need `IndexResult` at all after these changes?

**Comment by graetzer** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr-linux/3279/

**Comment by graetzer** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr-linux/3282/

**Comment by graetzer** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr-linux/3286/

**Pull request merged**